packages/main/src/SliderBase.js

```diff
@@ -0,0 +1,648 @@
+		this.fireEvent("input");
+	}
+
+	/**
+	 * Locks the given value between min and max boundaries based on slider properties
+	 *
+	 * @protected
+	 */
+	static clipValue(value, min, max) {
+		value = Math.min(Math.max(value, min), max);
+		return value;
+	}
+
+	/**
+	 * Sets the slider value from an event
+	 *
+	 * @protected
+	 */
+	static getValueFromInteraction(event, stepSize, min, max, boundingClientRect, directionStart) {
+		const pageX = this.getPageXValueFromEvent(event);
+		const value = this.computedValueFromPageX(pageX, min, max, boundingClientRect, directionStart);
+		const steppedValue = this.getSteppedValue(value, stepSize, min);
+
+		return this.clipValue(steppedValue, min, max);
+	}
+
+	/**
+	 * "Stepify" the raw value - calculate the new value depending on the specified step property
+	 *
+	 * @protected
+	 */
+	static getSteppedValue(value, stepSize, min) {
+		const stepModuloValue = Math.abs((value - min) % stepSize);
+
+		if (stepSize === 0 || stepModuloValue === 0) {
+			return value;
+		}
+
+		// Clip (snap) the new value to the nearest step
+		value = (stepModuloValue * 2 >= stepSize) ? (value + stepSize) - stepModuloValue : value - stepModuloValue;
+
+		// If the step value is not a round number get its precision
+		const stepPrecision = SliderBase._getDecimalPrecisionOfNumber(stepSize);
+		return value.toFixed(stepPrecision);
+	}
+
+	/**
+	 * Gets pageX value from event on user interaction with the Slider
+	 *
+	 * @protected
+	 */
+	static getPageXValueFromEvent(event) {
+		if (event.targetTouches && event.targetTouches.length > 0) {
+			return event.targetTouches[0].pageX;
+		}
+
+		return event.pageX;
+	}
+
+	/**
+	 * Computes the new value (in %) from the pageX position of the cursor.
+	 * Returns the value with rounded to a precision of at most 2 digits after decimal point.
+	 *
+	 * @protected
+	 */
+	static computedValueFromPageX(pageX, min, max, boundingClientRect, directionStart) {
+		// Determine pageX position relative to the Slider DOM
+		const xRelativePosition = directionStart === "left" ? pageX - boundingClientRect[directionStart] : boundingClientRect[directionStart] - pageX;
+		// Calculate the percentage complete (the "progress")
+		const percentageComplete = xRelativePosition / boundingClientRect.width;
+		// Fit (map) the complete percentage between the min/max value range
+		return min + percentageComplete * (max - min);
+	}
+
+	/**
+	 * Calculates the precision (decimal places) of a number, returns 0 if integer
+	 * Handles scientific notation cases.
+	 * @private
+	 */
+	static _getDecimalPrecisionOfNumber(value) {
+		if (Number.isInteger(value)) {
+			return 0;
+		}
+		const match = (String(value)).match(/(?:\.(\d+))?(?:[eE]([+-]?\d+))?$/);
+		return Math.max(0, (match[1] ? match[1].length : 0) - (match[2] ? Number(match[2]) : 0));
+	}
+
+	/**
+	 * Normalize current properties, update the previously stored state.
+	 *
+	 * @protected
+	 */
+	syncUIAndState(...values) {
```

Comment:
I guess this is an abstraction for the upcoming RangeSlider, which is very nice. Just again to reinforce, we should only change "value"-like properties, not min/max-like properties.